File: core/src/main/java/com/serious/service/exception/ServiceRuntimeException.java
package com.serious.service.exception;
/*
 * @COPYRIGHT (C) 2016 Andreas Ernst
 *
 * All rights reserved
 */

/**
 * @author Andreas Ernst
 */
public class ServiceRuntimeException extends RuntimeException {
    // ServiceException

    public ServiceRuntimeException(String message, Object... args) {
        super(String.format(message, args));
    }

    ServiceRuntimeException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: core/src/main/java/com/serious/service/exception/ServiceRuntimeException.kt
package com.serious.service.exception
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

/**
 * Any service related exception that occurs during the runtime.
 */
class ServiceRuntimeException : RuntimeException {
    constructor(message: String, vararg args: Any?) : super(String.format(message, *args))

    internal constructor(message: String, cause: Throwable?) : super(message, cause)
}